File: spec/cpp_stl/test_term_bytes.cpp
#include <boost/test/unit_test.hpp>

#include <term_bytes.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_term_bytes) {
    std::ifstream ifs("src/term_strz.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    term_bytes_t* r = new term_bytes_t(&ks);

    BOOST_CHECK_EQUAL(r->s1(), "foo");
    BOOST_CHECK_EQUAL(r->s2(), "bar");
    BOOST_CHECK_EQUAL(r->s3(), "|baz@");

    delete r;
}

File: spec/javascript/test_term_bytes.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('TermBytes', 'src/term_strz.bin', function(r) {
  assert.equal(KaitaiStream.bytesToStr(r.s1, "UTF-8"), 'foo');
  assert.equal(KaitaiStream.bytesToStr(r.s2, "UTF-8"), 'bar');
  assert.equal(KaitaiStream.bytesToStr(r.s3, "UTF-8"), '|baz@');
});
